packages/@lwc/engine/src/framework/wc.ts

```diff
@@ -25,17 +27,21 @@ import { HTMLElementConstructor } from './base-bridge-element';
  * const elm = document.createElement('x-foo');
  * ```
  */
-export function buildCustomElementConstructor(
-    Ctor: ComponentConstructor,
-    options?: {
-        mode?: 'open' | 'closed';
-    }
+export function deprecatedBuildCustomElementConstructor(
+    Ctor: ComponentConstructor
 ): HTMLElementConstructor {
+    /* eslint-disable-next-line no-console */
+    console.error(
+        'Deprecated function called: "buildCustomElementConstructor" function is deprecated and it will be removed in v1.4.0.' +
```

Comment:
Remove the expected build version when this will be dropped. We might readjust the version in the future if needed.

Reply:
I'm fine with that to be removed.